scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -221,17 +222,35 @@ class FossId(
                         .checkResponse("create project")
                 }
 
-                log.info { "Creating scan for URL $url and revision $revision ..." }
+                val scans = service.listScansForProject(user, apiKey, projectCode)
+                    .checkResponse("list scans for project").data
+                requireNotNull(scans)
 
-                val scanCode = createScan(projectCode, url, revision)
-
-                log.info { "Initiating download from Git repository ..." }
+                val existingScan = scans.sortedByDescending { scan -> scan.id }.find { scan ->
+                    scan.gitBranch == revision && scan.gitRepoUrl == url
+                }
 
-                service.downloadFromGit(user, apiKey, scanCode)
-                    .checkResponse("download data from Git", false)
+                val scanCode = kotlin.run {
```

Comment:
If I understand correctly, the `run` call is only necessary for the case that `existingScan` is *null*. Maybe this could be better expressed by changing the logic to something like:

```
existingScan?.let {
  log.info { "Scan found for url=$url and revision=$revision." }

  val scanCode = existingScan.code
  ...
} ?: kotlin.run {
  log.info { "No scan found for url=$url and revision=$revision. Creating scan ..." }
  val scanCode = createScan(projectCode, url, revision)
  ...
}

```

Reply:
Is this really more readable as a plain if..else ? :)

Reply:
Granted, this may be a matter of taste. My point was more that `run` should only be called if necessary. I would also be fine with moving this call inside the `if`.

Reply:
Yes, I did it regarding the other review.